providers/gce/gce_instances.go

```diff
@@ -121,12 +175,17 @@ func (g *Cloud) NodeAddresses(ctx context.Context, nodeName types.NodeName) ([]v
 				if err != nil {
 					return nil, fmt.Errorf("couldn't get internal IP: %v", err)
 				}
-				nodeAddresses = append(nodeAddresses, v1.NodeAddress{Type: v1.NodeInternalIP, Address: internalIP})
+
+				if internalIP == "" {
+					klog.Infof("No internal IPv4 address found for node %v.", nodeName)
+				} else {
+					nodeAddresses = append(nodeAddresses, v1.NodeAddress{Type: v1.NodeInternalIP, Address: internalIP})
+				}
```

Comment:
how is this change related?

Reply:
In new unit tests I added cases for single stack IPv6 VMs. Such VMs don't have internal IPv4 addresses, hence I had to make the IPv4 address optional.

Reply:
so this was a bug, was it adding an empty address?

Reply:
There was no test for a VM that would not have an internal IPv4 address. Most likely when the code was written, no such VMs could exist. It may be considered a bug, or just a new feature.